added tdarr widget

rshide opened the pull request:
Adding a widget for Tdarr to close [#404](https://github.com/benphelps/homepage/issues/404).

This will display 4 fields: 
- Count of items currently in queue
- Count of items that have been processed
- Count of items that have errored
- Total amount of disk space saved

Configuration example:
```        
widget:
    type: tdarr
    url: http://tdarr.host.or.ip
```

Preview:
![image](https://user-images.githubusercontent.com/32021884/208167589-7d39cbf4-e90a-46fe-b009-60cc53b3415b.png)

shamoon commented:
Thanks, we'll take a look. There are a couple comments in the code if you could remove but one actual thing is re. `headerCacheKey`, I dont see the purpose of this, its just setting the `content-type`? That shouldn't be necessary. And an odd API that the call is a `POST`

wazzirr commented:
Would be great to also see what's currently being transcoded, and time left

rshide commented:
> Thanks, we'll take a look. There are a couple comments in the code if you could remove but one actual thing is re. `headerCacheKey`, I dont see the purpose of this, its just setting the `content-type`? That shouldn't be necessary. And an odd API that the call is a `POST`

@shamoon I've removed the comments and the headerCacheKey as well.  Also added a preview and config example. I agree that this doesn't make much sense as a `POST`, but that is the only call that [their docs](https://tdarr.readme.io/reference/post_cruddb)  provide. Let me know if anything else needs to be cleaned up.


> Would be great to also see what's currently being transcoded, and time left

@wazzirr I think it makes sense to also allow time remaining to be displayed.  I can look into this more in the future and see if there is an API available for that.  

For what is being currently transcoded, are you thinking a list of files that are currently being transcoded similar to what is displayed under the Nodes section in Tdarr?  If so, are there any other widgets that currently display a list of items?

wazzirr commented:
Nvm, I forgot about an option to transcode more than one file at once.

shamoon commented:
Cheers thanks
